hajji/hello.S

```diff
@@ -49,8 +48,15 @@ start:
          * we anchor it at $LOAD.  Note that we don't collide with
          * the code because the stack will always remain below
          * (i.e. less than) $LOAD and grows downwards from there.
+         * disable intterupts when setting up the stack. If an
+         * interrupt occurs between the two MOVs then the stack
+         * may point at the wrong memory location and the interrupt
+         * may crash the system
          */
+        cli
```

Comment:
You can do it as the first line of the program however depending on what code someone is going to write they may want want to run with interrupts on for performance reasons. For small programs that don't use interrupts then turning them off at the start is fine. If you use interrupts it is good practice to keep them off for the minimum time necessary and re-enable them as quickly as possible.

Something else you should do is specifically when your program starts is call "CLD" if you intend to use any of the looping constructs. You can't guarantee the direction flag will be set properly.